kratos/utilities/element_size_calculator.cpp

```diff
@@ -116,6 +116,105 @@ double ElementSizeCalculator<2,3>::MinimumElementSizeDerivative(
     KRATOS_CATCH("");
 }
 
+// Triangle2D6 version.
+template<>
+double ElementSizeCalculator<2,6>::MinimumElementSize(const Geometry<Node<3> >& rGeometry)
+{
+
+    /* Calculate node-edge distances */
+    const double x10 = rGeometry[1].X() - rGeometry[0].X();
+    const double y10 = rGeometry[1].Y() - rGeometry[0].Y();
+    const double x20 = rGeometry[2].X() - rGeometry[0].X();
+    const double y20 = rGeometry[2].Y() - rGeometry[0].Y();
+
+    // node 0, edge 12
+    double nx = -(y20-y10);
+    double ny = x20-x10;
+    double Hsq = x10*nx + y10*ny;
+    Hsq *= Hsq / (nx*nx + ny*ny);
+
+    // node 1, edge 20
+    nx = -y20;
+    ny = x20;
+    double hsq = x10*nx + y10*ny;
+    hsq *= hsq / (nx*nx + ny*ny);
+    Hsq = ( hsq < Hsq ) ? hsq : Hsq;
+
+    // node 2, edge 10
+    nx = -y10;
+    ny = x10;
+    hsq = x20*nx + y20*ny;
+    hsq *= hsq / (nx*nx + ny*ny);
+    Hsq = ( hsq < Hsq ) ? hsq : Hsq;
+
+    return std::sqrt(Hsq);
+}
+// Triangle2D6 version.
+template<>
+double ElementSizeCalculator<2,6>::MinimumElementSizeDerivative(
+    const unsigned int DerivativeNodeIndex,
+    const unsigned int DerivativeDirectionIndex,
+    const Geometry<Node<3> >& rGeometry)
+{
+    KRATOS_TRY
+
```

Comment:
I'm not sure if the derivative is the same for quadratic geometries. @sunethwarna can you confirm please.

Besides, if the implementation is the same we must call the linear method.

Reply:
Hi, you are correct @rubenzorrilla. Derivatives are not the same. If you look at the following code snippet
https://github.com/KratosMultiphysics/Kratos/blob/f0b86d60271e58b4681800bb2b771596c7567d7c/kratos/utilities/element_size_calculator.cpp#L62-L64

The quadratice element has six nodes, and it only uses first three nodes for the `MinimumElementSize` calculation. It will give correct answers for the first three `DerivativeNodeIndex`, and will crash if the derivative w.r.t. forth or higher node index node is requested. Therefore I would have another templated method calling the `ElementSizeCalculator<2,3>::MinimumElementSizeDerivative` for first three nodes, and return zero for the rest of the nodes and their directions.

I think, if you run the test case in debug, it will crash due to the fact that error is produced only in debug mode.